drake/systems/plants/RigidBodySystem.h

```diff
@@ -388,61 +387,118 @@ class DRAKERBSYSTEM_EXPORT RigidBodySpringDamper
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 };
 
-/** RigidBodySensor
- * @brief interface class for elements which define a sensor which reads the
- * state of a rigid body system
+/** NoiseModel
+ * @brief Represents generalized vector-valued noise
  */
-class DRAKERBSYSTEM_EXPORT RigidBodySensor {
- public:
-  RigidBodySensor(RigidBodySystem* sys, const std::string& name)
-      : sys(sys), name(name) {}
-  virtual ~RigidBodySensor() {}
+template <typename ScalarType, int Dimension>
+class DRAKERBSYSTEM_EXPORT NoiseModel {
+public:
+    virtual Eigen::Matrix<ScalarType, Dimension, 1> generateNoiseVector() = 0;
+};
 
-  virtual size_t getNumOutputs() const { return 0; }
-  virtual Eigen::VectorXd output(
-      const double& t,
-      const KinematicsCache<double>& rigid_body_state) const = 0;
+/** GaussianNoiseModel
+ * @brief Implements the NoiseModel interface where the underlying noise distribution is parameterized by a Gaussian
+ */
+template <typename ScalarType, int Dimension>
+class DRAKERBSYSTEM_EXPORT GaussianNoiseModel : public NoiseModel<ScalarType, Dimension> {
+public:
+    GaussianNoiseModel(double mean, double std_dev) : distribution(mean, std_dev), generator(rd()) { }
+
+    virtual Eigen::Matrix<ScalarType, Dimension, 1> generateNoiseVector() override {
+        Eigen::Matrix<ScalarType, Dimension, 1> noise_vector;
+        for(std::size_t index = 0; index < Dimension; index++) {
+            noise_vector[index] = distribution(generator);
+        }
+        return noise_vector;
+    }
+private:
+    std::random_device rd;
+    std::normal_distribution<ScalarType> distribution;
+    std::mt19937 generator;
+};
 
- protected:
-  RigidBodySystem* sys;
-  std::string name;
+/** RigidBodySensor
+ * @brief interface class for elements which define a sensor which reads the state of a rigid body system
+ */
+class DRAKERBSYSTEM_EXPORT RigidBodySensor {
+public:
+    RigidBodySensor(RigidBodySystem* sys, const std::string& name) : sys(sys), name(name) {}
+    virtual ~RigidBodySensor() {}
+
+    virtual size_t getNumOutputs() const { return 0; }
+    virtual Eigen::VectorXd output(const double& t, const KinematicsCache<double>& rigid_body_state, const RigidBodySystem::InputVector<double>& u) const = 0;
+protected:
+    RigidBodySystem* sys;
```

Comment:
I think a raw pointer is not the right choice; should use either a reference or a shared_ptr here I think.